Registers with multiple reserved fields causes duplicate definitions

Running the tool on [this svd file](https://gist.github.com/simonvandel/6a67b7893f89482b8be2b7ac75a661b6) generates code that does not compile.

The error message reads:
```
error[E0201]: duplicate definitions with name `reserved`:
      --> src/lib.rs:156931:13
       |
156881 | /             pub fn reserved(&self) -> ReservedR {
156882 | |                 let bits = {
156883 | |                     const MASK: u8 = 1;
156884 | |                     const OFFSET: u8 = 0;
...      |
156887 | |                 ReservedR { bits }
156888 | |             }
       | |_____________- previous definition of `reserved` here
...
156931 | /             pub fn reserved(&self) -> ReservedR {
156932 | |                 let bits = {
156933 | |                     const MASK: u32 = 134217727;
156934 | |                     const OFFSET: u8 = 5;
...      |
156937 | |                 ReservedR { bits }
156938 | |             }
       | |_____________^ duplicate definition

```

This happens as a lot of registers in the svd file has multiple reserved fields.

I see two different solutions to this problem:
1. Edit the SVD file, appending a unique string to each conflicting RESERVED field.
2. Change svd2rust, by special casing registers  with multiple reserved fields such that unique field names are generated. i.e. RESERVED1, RESERVED2

What do you think?

--- a/src/generate.rs
+++ b/src/generate.rs
@@ -555,9 +555,14 @@ pub fn register(
     );
 
     if let Some(fields) = register.fields.as_ref() {
+        // filter out all reserved fields, as we should not generate code for them
+        let fields:Vec<Field> = fields.clone().into_iter()
+            .filter(|field| field.name.to_lowercase() != "reserved")
+            .collect();
+
         if !fields.is_empty() {
             ::generate::fields(
-                fields,
+                &fields,
                 register,
                 all_registers,
                 peripheral,
@@ -712,6 +717,8 @@ pub fn fields(
                 let has_reserved_variant = evs.values.len() != (1 << f.width);
                 let variants = evs.values
                     .iter()
+                    // filter out all reserved variants, as we should not generate code for them
+                    .filter(|field| field.name.to_lowercase() != "reserved")
                     .map(|ev| {
                         let sc =
                             Ident::new(&*ev.name.to_sanitized_snake_case());
@@ -1025,6 +1032,8 @@ pub fn fields(
 
                 let variants = evs.values
                     .iter()
+                    // filter out all reserved variants, as we should not generate code for them
+                    .filter(|field| field.name.to_lowercase() != "reserved")
                     .map(
                         |ev| {
                             let value = u64(ev.value.ok_or_else(|| {

